repos/system_upgrade/el7toel8/actors/storagescanner/libraries/library.py

```diff
@@ -79,22 +80,46 @@ def _get_fstab_info(fstab_path):
     ''' Collect storage info from /etc/fstab file '''
     if _is_file_readable(fstab_path):
         with open(fstab_path, 'r') as fstab:
-            for entry in fstab:
+            keys = ['fs_spec', 'fs_file', 'fs_vfstype', 'fs_mntops', 'fs_freq', 'fs_passno']
+            for line, entry in enumerate(fstab, 1):
                 if entry.startswith('#'):
                     continue
 
+                if '#' in entry:
+                    api.current_logger().warning(
+                        'Detected in /etc/fstab a "#" comment in line %s please verify if it is correct.',
+                        line
+                    )
+
                 entry = entry.strip()
                 if not entry:
                     continue
 
-                fs_spec, fs_file, fs_vfstype, fs_mntops, fs_freq, fs_passno = entry.split()
-                yield FstabEntry(
-                    fs_spec=fs_spec,
-                    fs_file=fs_file,
-                    fs_vfstype=fs_vfstype,
-                    fs_mntops=fs_mntops,
-                    fs_freq=fs_freq,
-                    fs_passno=fs_passno)
+                entries = entry.split()
+
+                if len(entries) not in [4, 5, 6]:
+                    summary = (
+                        'The fstab configuration file is invalid, you need to fix this to avoid'
+                        'problems in the upgrade process.'
+                    )
+                    remediation = (
+                        'The file /etc/fstab has a problem on the line: {}, it must have'
+                        'at least 4 values and at most 6, this line has {}. For futher information please read '
+                        'fstab man page (man 5 fstab)'.format(line, len(entries))
+                    )
+                    reporting.report_with_remediation(
+                        title='Possible problem with mount points',
+                        summary=summary,
+                        remediation=remediation,
+                        severity='high',
+                        flags=['inhibitor']
+                    )
+
+                    api.current_logger().error(summary)
+                    continue
+
+                data = dict(zip(keys, entries))
+                yield FstabEntry(**data)
```

Comment:
I have 2 concerns here - kwargs are against our guidelines unless absolutely necessary and data isn't validated anywhere (on the other side - it has never been)